### DISCUSSION TITLE
React app not getting deployed to https

### DISCUSSION THREAD

**Discussion by angelin-mary** [community/community#106614] [Programming Help]
### Body

I tried to deploy my react app with nginx. It is not getting deployed to domain with https. 
my configuration file is:
server {
    listen       443;
    server_name  hostname;

    #access_log  /var/log/nginx/host.access.log  main;

    location / {
        #root   /usr/share/nginx/html;
        #index  index.html index.htm;
         proxy_pass   http://127.0.0.1:5000;
    }

    error_page   500 502 503 504  /50x.html;
    location = /50x.html {
        root   /usr/share/nginx/html;
    }
I restarted nginx with this and started pm2. Still the site is not getting up in the domain.
Please help me with this issue

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sapalazz** [comment]
Hello @angelin-mary! Hope you're well.

Let me try to help you.

I think your Nginx configuration file needs more info to be set up. It seems to be missing SSL-related directives. Here's a basic example of how to configure SSL in Nginx:
```

server {
    listen 443 ssl;
    server_name hostname;

    ssl_certificate /path/to/your/certificate.crt;
    ssl_certificate_key /path/to/your/private.key;

    location / {
        proxy_pass http://127.0.0.1:5000;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }

    error_page 500 502 503 504 /50x.html;
    location = /50x.html {
        root /usr/share/nginx/html;
    }
}
```

Replace these two 

`/path/to/your/certificate.crt` & `/path/to/your/private.key`

with the paths to your SSL certificate and private key files respectively.

And of course, check that those SSL certificates are correct and valid for your purposes. After making those changes, restart Nginx to apply them. 

Let me know if that helped, hope it did. Have a nice day!

**Comment by angelin-mary** [comment]
Hi @sapalazz 
Thankyou for your response.

When I contacted my company about SSL_certificates this is the response I got.
" please setup your service using HTTP only.  Since requests to the service 
are going via proxy https://domainname[:443]/tprs, the communication between
client and DFKI-Webproxy is safe.  The last step between (company) proxy and your
service is within two (company) networks inhouse, thus it will be safe enough
without extra encryption "
My site is getting up in http://hostname/tprs when I use port 80 in the configuration file. But when i use 443 is not not getting connected. 
What I want is to get the site up in www.domainname/tprs.

Please help me with this if you can.
Thankyou

**Comment by smart-dev420** [comment]
I attached the code and it worked fine in my project.
```
server {
	
	listen 443 ssl default_server;
	root /var/www/html;

	# Add index.php to the list if you are using PHP
	index index.html index.htm index.nginx-debian.html;

	server_name mysite.io www.mysite.io;

	location / {
		# First attempt to serve request as file, then
		# as directory, then fall back to displaying a 404.
		proxy_pass http://127.0.0.1:3000/;

		proxy_http_version 1.1;
	    proxy_set_header X-Real-IP $remote_addr;
	    proxy_set_header Upgrade $http_upgrade;
	    proxy_set_header Connection 'upgrade';
	    proxy_set_header Host $host;
	    proxy_cache_bypass $http_upgrade;
	}

	location /_next/static {
	    alias /var/tokenarius/tokenarius.io/apps/furo/.next/static;
	    add_header Cache-Control "public, max-age=3600, immutable";
	}

	error_page 502 /502.html;
    location /502.html {
        root /var/www/html;
    }
}
```
